### ISSUE TITLE
Tabs Displaying in Wrong Location

### ISSUE THREAD

**Issue by MickiTee** [MrAlex94/Waterfox#2362]
### What happened?

A bug happened! I have two user accounts on my Windows 10 Home laptop - one for personal use and one for work. Since the update to Waterfox V4.0.0.1 the tabs in my work account are appearing at the bottom of the screen (see attached) despite the setting showing that "top below address bar" (see attached). This is NOT happening in my personal user account.

I have compared the tab settings in Waterfox on both accounts and they are identical so I do not understand why Waterfox in my work account is not displaying correctly.

Please find attached copies of Excel spreadsheets which contain the data from about:support and about:config for both users.

![Waterfox 4 0 0 1 Screenshot](https://user-images.githubusercontent.com/18121158/138686904-d8c13ab2-b185-4a63-8ab3-c7b4322d06eb.jpg)
![Waterfox 4 0 0 1 Settings - Tabs](https://user-images.githubusercontent.com/18121158/138686909-25c1cacc-cb53-4823-b14c-27f0df07521b.jpg)
[Waterfox Setting - Personal.xlsx](https://github.com/WaterfoxCo/Waterfox/files/7409622/Waterfox.Setting.-.Personal.xlsx)
[Waterfox Settings 20211021.xlsx](https://github.com/WaterfoxCo/Waterfox/files/7409624/Waterfox.Settings.20211021.xlsx)

Please let me know if you require any further information.

### Reproducible?

- [ ] I have tested this on Firefox.

### Version

G4.𝑋 (Future)

### What platform are you seeing the problem on?

Windows

**Comment by adamp01** [comment]
This certainly is a weird one - I can confirm your prefs have been updated correctly, so I'll need to try and replicate this to delve into it further.

**Comment by Objectivite** [comment]
> 
> ### What happened?
> 
> A bug happened! I have two user accounts on my Windows 10 Home laptop - one for personal use and one for work. Since the update to Waterfox V4.0.0.1 the tabs in my work account are appearing at the bottom of the screen (see attached) despite the setting showing that "top below address bar" (see attached). This is NOT happening in my personal user account.
> 
> I have compared the tab settings in Waterfox on both accounts and they are identical so I do not understand why Waterfox in my work account is not displaying correctly.
> 
> Please find attached copies of Excel spreadsheets which contain the data from about:support and about:config for both users.
> 
> ![Waterfox 4 0 0 1 Screenshot](https://user-images.githubusercontent.com/18121158/138686904-d8c13ab2-b185-4a63-8ab3-c7b4322d06eb.jpg) ![Waterfox 4 0 0 1 Settings - Tabs](https://user-images.githubusercontent.com/18121158/138686909-25c1cacc-cb53-4823-b14c-27f0df07521b.jpg) [Waterfox Setting - Personal.xlsx](https://github.com/WaterfoxCo/Waterfox/files/7409622/Waterfox.Setting.-.Personal.xlsx) [Waterfox Settings 20211021.xlsx](https://github.com/WaterfoxCo/Waterfox/files/7409624/Waterfox.Settings.20211021.xlsx)
> 
> Please let me know if you require any further information.
> ### Reproducible?
> 
>     * [ ]  I have tested this on Firefox.
> 
> 
> ### Version
> 
> G4.𝑋 (Future)
> ### What platform are you seeing the problem on?
> 
> Windows
> ### Relevant log output
> 
> _No response_

This is exactly the same problem I reported on Day 1, which many others also reported, and which was dismissed out of hand by the programmers as a Theme problem.  I even got a reply :  **Just change your theme** !!!   The same things that work in Firefox without problems.

In the end backed up AppData\Roaming\Waterfox  entire structure, uninstalled Waterfox, deleted all Waterfox keys from HKEY_CURRENT_USER, deleted the Waterfox program installation folder, deleted the AppData\Roaming\Waterfox structure, re-installed, imported the correct profile and things worked properly as regards the Tab bar.  It was never a theme problem of course.

If you do that, it will work - have done it 3 times, but you run into closed websites, and lost passwords, but at least you have the Tab bar where you want it and operating normally.

Good Luck.

O.

**Comment by MrAlex94** [comment]
> This is exactly the same problem I reported on Day 1, which many others also reported, and which was dismissed out of hand by the programmers as a Theme problem. I even got a reply : Just change your theme !!! The same things that work in Firefox without problems.

That doesn't seem related to this issue? Firefox does not support switching tab positioning and tab positions should be unrelated to theming, unless custom CSS is applied relating to tabs.

@MickiTee thank you for the detailed report - we'll try to reproduce this (so far unsuccessful on G3 -> G4 tests) so we can debug it.

**Comment by adamp01** [comment]
@MickiTee - would you be able to upload a log of the browser console (ctrl+shift+j) output after starting up a new window that displays this issue?

**Comment by MickiTee** [comment]
Any progress on resolving this issue? Since the latest Waterfox update to V4.0.2.1 the situation has deteriorated. I can now only see the active tab at the bottom of the screen. None of the others.

**Comment by Objectivite** [comment]
> 
> 
> Any progress on resolving this issue? Since the latest Waterfox update to V4.0.2.1 the situation has deteriorated. I can now only see the active tab at the bottom of the screen. None of the others.

You won't get anywhere (a good read of the thread will show that).

Do what I did :  take a copy of your profile folder (Help / Troubleshooting information will show you), uninstall, delete all folders from **Program Files** and **<User>\AppData\Roaming\Waterfox**, delete all registry keys under  HKEY_LOCAL_MACHINE\SOFTWARE\Waterfox (all flavours)  and HKEY_CURRENT_USER\SOFTWARE\Waterfox (all flavours), re-install Waterfox and then restore your backed up favourites from the backed up profile folder.  It does mean you will have to re-do all your cookies and saved passwords, but at least the toolbars and tabs will be back working.

In other words rely on yourself to fix what was broken through no fault of your own.  Not a single one of the 4.x updates will fix this for you.

O.

**Comment by adamp01** [comment]
This should now be resolved as of G4.0.4, please re-open the issue if that is not the case for you.

Thanks

**Comment by Objectivite** [comment]
Thanks Adam - it is.

O.
